### PR TITLE
chore: update macOS runner image versions

### PR DESCRIPTION
**Pull request by austinvazquez**
*Issue #, if available:*
https://github.com/runfinch/finch/pull/949 is currently blocked due to virtualization framework issues on macOS 13.2.

*Description of changes:*
This change updates the macOS runners to macOS 13.6 and macOS 14.5

*Testing done:*

- [x] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: config/runner-config.json [hunk 1]
```diff
@@ -129,79 +129,47 @@
...
-        "availabilityZones": ["us-east-2a", "us-east-2b", "us-east-2c"]
-      },
-      {
-        "platform": "mac",
-        "version": "11.7",
-        "arch": "x86",
-        "repo": "finch",
-        "desiredInstances": 1,
-        "availabilityZones": ["us-east-2a", "us-east-2b", "us-east-2c"]
-      },
```
Comment (austinvazquez) [inline config/runner-config.json:177]: Let's remove macOS 11 runners as it is EOL.

Comment (pendo324) [reply config/runner-config.json:177]: I don't believe we're using these anywhere in our GitHub workflows in runfinch/finch or runfinch/finch-core, but can you double check?

**Review by pendo324 — COMMENTED** [review]
Change LGTM, just want to make sure we're not using the macOS 11 runners before we unprovision them

File: config/runner-config.json (hunk 1 above)
Comment (austinvazquez) [reply config/runner-config.json:177]: Yes, good catch. Let me undo this and follow-up seperately.

https://github.com/runfinch/finch/blob/17d4bc8fbdf6b80e5070d2aa3ed1c85b112252fb/.github/workflows/build-and-test-pkg.yaml#L59

Comment (pendo324) [reply config/runner-config.json:177]: Sounds good. FWIW, I think we can remove those macOS 11 runners safely. But we don't have to tie it all into this PR so I agree with rolling it back and taking it as followup.

**Review by pendo324 — APPROVED** [review]

**Pull request merged**